### DISCUSSION TITLE
Disable strict input type coercion for a single query?

### DISCUSSION THREAD

**Discussion by hossein** [rebing/graphql-laravel#917] [Q&A] [answered]
We have a specific GraphQL query accepting a specific nullable `String` input. Our GraphQL client app has an issue that sends `false` instead of `null` strings to this query, which stopped working since webonyx 14 due to implementing [#278 Spec Compliance: Strict input coercion for scalars](https://github.com/webonyx/graphql-php/issues/278). However, we cannot update the client apps easily because they are on users' machines that we don't control (not web-based).

Question: Is there a way to hook early into the code and allow/convert the invalid input values for this specific query, avoiding generating the error `Variable "$var" got invalid value false; Expected type String; String cannot represent a non string value: false`?

Using GraphQL middleware didn't work (types already validated before middleware are run) and I'm trying to avoid HTTP middleware to not impose overhead to the entire app because of this single issue.

**Comment by mfn** [comment, marked as answer]
I remember also being impacted by the strict input coercion. The previous way felt more "PHP style" to me, but that ship has sailed anyway.

I'm sorry I don't have anymore recommendations, adding a Laravel HTTP middleware would be my solution.

I know you evaluating that option and I cannot know your stack, but have you already benchmarked such a solution? In my experience the GraphQL type system overhead (building the schema, parsing the request, validating all the fields in the response) brings more latency to a request than a middleware manipulating some data structure slightly.

**Reply by hossein** [reply]
You're right, an HTTP middleware operating only on graphql that searches the query name in the first characters of the request body should have acceptable performance. Thanks for the feedback.
